llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1155,6 +1155,11 @@ RISCVTTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
     return getCmpSelInstrCost(Instruction::Select, ICA.getReturnType(),
                               ICA.getArgTypes()[0], CmpInst::BAD_ICMP_PREDICATE,
                               CostKind);
+  case Intrinsic::experimental_vp_splat: {
+    auto LT = getTypeLegalizationCost(RetTy);
```

Comment:
You're missing the hasV check, and it looks like this is only handling integer (not FP).

Reply:
And also the i1 case

Reply:
If there's no V instructions then `getTypeLegalizationCost` will return an invalid type legalization cost for scalable vectors, or a scalar legalized type for fixed vectors, both of which will end up in invalid cost. We could be more explicit about this though?

Reply:
The fixed vector case will probably end up passing a scalar type to the getRISCVInstructionCost routine which I don't think is what we want.  Please add the same type of guard we see elsewhere in this switch for the moment, we can come back and revisit in batch.

Reply:
Sounds good. I've made it return invalid rather than falling out of the switch since we can't scalarize scalable nor fixed versions of this intrinsic yet

Reply:
As a follow up, can you add handling for fixed vectors (only) in BasicTTI via getShuffleCost?  We should be modeling the scalarization here.

Reply:
I took a look into this, but ended up in a bit of a rabbit hole. The default BasicTTIImpl scalarization implementation calls into the InsertElement/ExtractElement cost, but currently we return a cost of 0 for constant indices which at least isn't true for the inserts generated from a scalarized splat shuffle, for example.

However changing the InsertElement/ExtractElement cost seems somewhat sensitive, see #67334. I think we may want to make the cost "dumber", e.g. just a single scalar load/store for any insert or extract, but we would need to double check this doesn't cause excessive unrolling.

This is more complicated than what I first thought it would be, so leaving a note here to maybe return to later.